pkg/webhook/webhook.go

```diff
@@ -55,15 +59,49 @@ func (q *QuayIntegrationMutator) Handle(ctx context.Context, req admission.Reque
 			}
 		}
 	} else {
-
-		admissionResponse = getAdmissionResponseForBuild(build, &quayIntegration)
+		// Check if builder service account has secret
+		hasSecret, serviceAcctErr := q.checkSecretForBuilderServiceAccount(ctx, &req, &quayIntegration)
+
+		if !hasSecret {
+			if serviceAcctErr != nil {
+				admissionResponse = &admissionv1.AdmissionResponse{
+					Allowed: false,
+					Result: &metav1.Status{
+						Message: serviceAcctErr.Error(),
+					},
+				}
+			} else {
+				admissionResponse = &admissionv1.AdmissionResponse{
+					Allowed: false,
+				}
+			}
+		} else {
+			admissionResponse = getAdmissionResponseForBuild(build, &quayIntegration)
+		}
 
 	}
 
 	return admission.Response{AdmissionResponse: *admissionResponse}
 
 }
 
+func (q *QuayIntegrationMutator) checkSecretForBuilderServiceAccount(ctx context.Context, ar *admission.Request, quayIntegration *quayv1.QuayIntegration) (bool, error) {
+
```

Comment:
Just curious, why do you add empty line between the function signature and its implementation?

Reply:
For better readability (and being consistent to how other functions are formatted). I have removed it since it's a small helper func